src/gtk/textctrl.cpp

```diff
@@ -1017,6 +1023,57 @@ void wxTextCtrl::GTKSetJustification()
     }
 }
 
+#if wxUSE_SPELLCHECK
+
+bool wxTextCtrl::EnableProofCheck(bool enable, const wxTextProofOptions& options)
+{
+    wxCHECK_MSG( IsMultiLine(), false,
+                "Unable to enable spell check on control "
+                "which does not have wxTE_MULTILINE style" );
+
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+    wxCHECK_MSG( textview, false, wxS("wxTextCtrl is not a GtkTextView"));
+
+    GtkSpellChecker *spell = gtk_spell_checker_get_from_text_view(textview);
+
+    if ( enable )
+    {
+        if ( !spell )
+        {
+            spell = gtk_spell_checker_new();
+            gtk_spell_checker_attach(spell, textview);
+        }
+
+        wxString lang = options.GetLang();
+
+        if ( !gtk_spell_checker_set_language(spell,
+                                    lang.IsEmpty() ? NULL : (const gchar *)lang.c_str(),
```

Comment:
```suggestion
                                    lang.empty() ? NULL : (const gchar *)lang.utf8_str(),
```
First change is just a matter of taste (I think standard-like methods should be preferred), but the second one might be more important if we ever have languages using non-ASCII characters (currently it can't happen AFAIK but why not support them if it ever does, as long as it doesn't cost anything).

Reply:
I will make these updates on my local copy to avoid a merge (I've implemented `=  wxTextProofOptions()` locally).